src/rulesets/LinearAlgebra/uniformscaling.jl

```diff
@@ -0,0 +1,93 @@
+
+#####
+##### constructor
+#####
+
+function rrule(::Type{T}, x::Number) where {T<:UniformScaling}
+    UniformScaling_back(dx) = (NoTangent(), ProjectTo(x)(unthunk(dx).λ))
+    return T(x), UniformScaling_back
+end
+
+#####
+##### `+`
+#####
+
+function frule((_, Δx, ΔJ), ::typeof(+), x::AbstractMatrix, J::UniformScaling)
+    return x + J, Δx + ΔJ
+end
+
+function frule((_, ΔJ, Δx), ::typeof(+), J::UniformScaling, x::AbstractMatrix)
+    return J + x, ΔJ + Δx
+end
+
+function rrule(::typeof(+), x::AbstractMatrix, J::UniformScaling)
+    project_x = ProjectTo(x)
+    project_J = ProjectTo(J)
+    function plus_back(dy)
+        dx = unthunk(dy)
+        (NoTangent(), project_x(dx), project_J(I * tr(dx)))
+    end
+    return x + J, plus_back
+end
+
+function rrule(::typeof(+), J::UniformScaling, x::AbstractMatrix)
+    y, back = rrule(+, x, J)
+    function plus_back_2(dy)
+        df, dx, dJ = back(dy)
+        return (df, dJ, dx)
+    end
+    return y, plus_back_2
+end
+
+#####
+##### `-`
+#####
+
+function frule((_, Δx, ΔJ), ::typeof(-), x::AbstractMatrix, J::UniformScaling)
+    return x - J, Δx - ΔJ
+end
+
+function frule((_, ΔJ, Δx), ::typeof(-), J::UniformScaling, x::AbstractMatrix)
+    return J - x, ΔJ - Δx
+end
+
+function rrule(::typeof(-), x::AbstractMatrix, J::UniformScaling)
+    y, back = rrule(+, x, -J)
+    function minus_back_1(dy)
+        df, dx, dJmaybe = back(dy)
+        dJ = J.λ isa Bool ? NoTangent() : dJmaybe  # as -true isa Int
+        return (df, dx, -dJ)
+    end
+    return y, minus_back_1
+end
+
+function rrule(::typeof(-), J::UniformScaling, x::AbstractMatrix)
+    project_x = ProjectTo(x)
+    project_J = ProjectTo(J)
+    function minus_back_2(dy)
+        dx = -unthunk(dy)
```

Comment:
```suggestion
        dx = -dy
```
`-` works on thunks right?

Reply:
Apparently. But I think the intention is clearer with it written out. Otherwise it's tempting to save a line, which it less efficient but tests won't tell you.